style/src/rule.rs

```diff
@@ -0,0 +1,51 @@
+//! Display a horizontal or vertical rule for dividing content.
+
+use iced_core::Color;
+
+/// The appearance of a rule.
+#[derive(Debug, Clone, Copy)]
+pub struct Style {
+    /// The color of the rule.
+    pub color: Color,
+    /// The width (thickness) of the rule line.
+    pub width: u16,
+    /// The radius of the rectangle corners.
+    pub radius: u16,
+    /// The percent from [0, 100] of the filled space the rule
+    /// will be drawn.
+    pub fill_percent: u16,
```

Comment:
A few things here:

- Why not either `f32` or `u8`? The upper byte of a `u16` will never be used for values on `[0, 100]`.
- What about a `FillMode` enum?

```rust
enum FillMode {
    // Current behavior
    Percent(f32),
    // Uniform offset from each end, length units
    Padded(f32),
    // Different offset on each end of the rule. Start = top  or left
    AsymmetricPadding { start: f32, end: f32 }, 
}
```

Reply:
I like the idea of the `FillMode` enum. I'll go ahead and implement that.

As for the u16, all the other styles in iced use u16, so I thought I would keep it consistent. It would make sense for `Percent` to be floating point.

In iced's other stylesheets, anything that deals with units (pixels) is always u16. Maybe this could be changed once iced gets a better sub-pixel anti-aliasing system.

Reply:
What are the use cases for `AsymetricPadding`? I would try to keep it simple for now.

Reply:
Plenty of aesthetic reasons to want asymmetry. You might want the start of the line to be padded and the end to run-out the full container. I'm sure you could achieve the same effect by wrapping a rule in other containers with spacers, padding, etc, but that is more complex than first-class support in a single element.